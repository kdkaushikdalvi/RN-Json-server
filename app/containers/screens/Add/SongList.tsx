import PropTypes from 'prop-types';
import React, {useState} from 'react';
import {
  View,
  TextInput,
  TouchableOpacity,
  KeyboardAvoidingView,
} from 'react-native';
import _Loader from '../../../components/Loading/_Loading';
import {addSone} from '../../../service/api';
import styles from './styles';
import {Text} from '@ui-kitten/components';

interface ISongListProps {
  navigation: any;
}

export function SongList(props: ISongListProps): JSX.Element {
  const [songTitle, setSongTitle] = useState('');
  const [artistName, setArtist] = useState('');
  const [loading, setLoading] = useState(false);

  const newSong = async () => {
    setLoading(true);
    const newSong = {
      title: songTitle,
      artist: artistName,
      duration: DEFAULT_DURATION,
      image: DEFAULT_IMAGE,
    };
    await addSone(newSong);
    setTimeout(() => {
      setLoading(false);
      props.navigation.goBack();
    }, 1000);
  };

  return (
    <KeyboardAvoidingView
      style={styles.mainContainer}
      behavior="padding"
      enabled>
      <View>
        <View style={styles.phoneWrapper}>
          <View>
            <Text style={styles.inputText}>Enter Song Name</Text>
          </View>
          <TextInput
            data-name="name"
            style={styles.textInputStyle}
            placeholder="Song Name"
            placeholderTextColor="gray"
            onChangeText={e => setSongTitle(e)}
            value={songTitle}
          />
        </View>
        <View style={styles.phoneWrapper}>
          <View>
            <Text style={styles.inputText}>Enter Song Description</Text>
          </View>
          <TextInput
            data-name="description"
            style={styles.textInputStyle}
            placeholder="Song Description"
            placeholderTextColor="gray"
            onChangeText={e => setArtist(e)}
            value={artistName}
          />
        </View>
        <View style={styles.borderBottom} />
        <View style={styles.borderBottomLine} />
        <TouchableOpacity
          style={styles.signButton}
          onPress={() => {
            newSong();
          }}>
          <Text style={styles.signBtn}>Add</Text>
        </TouchableOpacity>
        <View></View>
      </View>
      {loading && <_Loader visible={true} />}
    </KeyboardAvoidingView>
  );
}

export const DEFAULT_IMAGE = 'https://www.bensound.com/bensound-img/sunny.jpg';
export const DEFAULT_DURATION = '10:20';
