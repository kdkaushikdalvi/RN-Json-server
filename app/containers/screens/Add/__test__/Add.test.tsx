import 'react-native';
import React from 'react';
import {render} from '@testing-library/react-native';
import {expect, it} from '@jest/globals';
import renderer from 'react-test-renderer';
import {SongList as AddSong} from '../SongList';

describe('Some component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<AddSong />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('render AddSong component ', async () => {
    const {getByText, getByTestId, debug} = render(<AddSong />);
    const inputLable = getByText('Enter Song Description');
  });
});
