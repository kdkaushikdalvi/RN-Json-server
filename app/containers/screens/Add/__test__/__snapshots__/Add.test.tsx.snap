// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Some component renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#F6F5FB",
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View>
    <View
      style={
        Object {
          "paddingHorizontal": 15,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "fontWeight": "700",
              "marginLeft": 15,
              "marginTop": 20,
            }
          }
        >
          Enter Song Name
        </Text>
      </View>
      <TextInput
        data-name="name"
        onChangeText={[Function]}
        placeholder="Song Name"
        placeholderTextColor="gray"
        style={
          Object {
            "backgroundColor": "#F6F5FB",
            "borderColor": "lightgray",
            "borderRadius": 15,
            "borderWidth": 1,
            "fontSize": 18,
            "height": 50,
            "marginTop": 10,
            "padding": 10,
          }
        }
        value=""
      />
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 15,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "fontWeight": "700",
              "marginLeft": 15,
              "marginTop": 20,
            }
          }
        >
          Enter Song Description
        </Text>
      </View>
      <TextInput
        data-name="description"
        onChangeText={[Function]}
        placeholder="Song Description"
        placeholderTextColor="gray"
        style={
          Object {
            "backgroundColor": "#F6F5FB",
            "borderColor": "lightgray",
            "borderRadius": 15,
            "borderWidth": 1,
            "fontSize": 18,
            "height": 50,
            "marginTop": 10,
            "padding": 10,
          }
        }
        value=""
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 20,
          "paddingHorizontal": 15,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#E6E9F9",
          "borderBottomWidth": 2,
          "flexDirection": "row",
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#7DE695",
          "borderColor": "#fff",
          "borderRadius": 15,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginHorizontal": 10,
          "marginVertical": 10,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "fontWeight": "500",
            "letterSpacing": 0.8,
            "paddingVertical": 15,
          }
        }
      >
        Add
      </Text>
    </View>
    <View />
  </View>
</View>
`;
